File: Forge/src/main/java/com/example/examplemod/mixin/ExampleMixin.java
package com.example.examplemod.mixin;

import com.example.examplemod.Constants;
import net.minecraft.server.MinecraftServer;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.injection.At;
import org.spongepowered.asm.mixin.injection.Inject;
import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;

@Mixin(MinecraftServer.class)
public class ExampleMixin {
	@Inject(at = @At("HEAD"), method = "loadLevel")
	private void init(CallbackInfo info) {
		Constants.LOGGER.info("This line is printed by a forge mixin!");
	}
}

File: MultiLoader-Template-with-kotlin-and-kotlin-dsl/Fabric/src/main/kotlin/com/example/examplemod/mixin/ExampleMixin.kt
package com.example.examplemod.mixin

import com.example.examplemod.Constants
import net.minecraft.client.Minecraft
import net.minecraft.client.gui.screens.TitleScreen
import org.spongepowered.asm.mixin.Mixin
import org.spongepowered.asm.mixin.injection.At
import org.spongepowered.asm.mixin.injection.Inject
import org.spongepowered.asm.mixin.injection.callback.CallbackInfo

@Mixin(TitleScreen::class)
class ExampleMixin {
    @Inject(at = [At("HEAD")], method = ["init()V"])
    private fun init(info: CallbackInfo) {
        Constants.LOG.info("This line is printed by an example mod mixin from Fabric!")
        Constants.LOG.info("MC Version: {}", Minecraft.getInstance().versionType)
        Constants.LOG.info("Classloader: {}", this.javaClass.classLoader)
    }
}